src/app/auth/api.http.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { ConnectionBackend, Http, Headers, Request, RequestOptions, RequestOptionsArgs, Response } from '@angular/http';
+import { Observable } from "rxjs";
+
+import { apiToken } from "../constants";
+
+import * as _ from 'lodash';
+
+/**
+ * Wrapper for Http that appends authorization headers for requests to the API backend,
+ * and redirects unauthorized responses to login page.
+ */
+@Injectable()
+export class ApiHttp extends Http {
+
+    constructor(protected _backend: ConnectionBackend, protected _defaultOptions: RequestOptions) {
+        super(_backend, _defaultOptions);
+    }
+
+    request(url: string | Request, options?: RequestOptionsArgs): Observable<Response> {
+        return this.intercept(super.request(url, this.appendAuthHeader(options)));
+    }
+
+    get(url: string, options?: RequestOptionsArgs): Observable<Response> {
+        return this.intercept(super.get(url, this.appendAuthHeader(options)));
+    }
+
+    post(url: string, body: string, options?: RequestOptionsArgs): Observable<Response> {
+        return this.intercept(super.post(url, body, this.appendAuthHeader(options)));
+    }
+
+    put(url: string, body: string, options?: RequestOptionsArgs): Observable<Response> {
+        return this.intercept(super.put(url, body, this.appendAuthHeader(options)));
+    }
+
+    delete(url: string, options?: RequestOptionsArgs): Observable<Response> {
+        return this.intercept(super.delete(url, options));
+    }
+
+    appendAuthHeader(options?: RequestOptionsArgs) : RequestOptionsArgs {
+        if (options == null) {
+            options = new RequestOptions();
+        }
+        if (options.headers == null) {
+            options.headers = new Headers();
+        }
+        options.headers.set('Authorization', 'Token ' + apiToken);
```

Comment:
Rather than import constants directly here, could we inject it at app bootstrap using providers? This would allow us to reuse this class in other projects.

See: https://angular.io/docs/ts/latest/guide/dependency-injection.html#!#dependency-injection-tokens